src/material/list/testing/list-item-harness.ts

```diff
@@ -0,0 +1,248 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ComponentHarness,
+  ComponentHarnessConstructor,
+  HarnessLoader,
+  HarnessPredicate
+} from '@angular/cdk/testing';
+import {
+  MAT_ACTION_LIST_SELECTOR,
+  MAT_LIST_SELECTOR, MAT_NAV_LIST_SELECTOR
+} from '@angular/material/list/testing/list-selectors';
+import {
+  ActionListItemHarnessFilters,
+  BaseListItemHarnessFilters,
+  ListItemHarnessFilters,
+  ListOptionHarnessFilters,
+  NavListItemHarnessFilters,
+  SubheaderHarnessFilters
+} from './list-harness-filters';
+
+function getListItemPredicate
+    <H extends MatListItemHarnessBase, F extends BaseListItemHarnessFilters>
+    (harnessType: ComponentHarnessConstructor<H>, options: F): HarnessPredicate<H> {
+  return new HarnessPredicate(harnessType, options)
+      .addOption('text', options.text,
+          (harness, text) => HarnessPredicate.stringMatches(harness.getText(), text));
+}
+
+/** Harness for interacting with a list subheader. */
+export class MatSubheaderHarness extends ComponentHarness {
+  static hostSelector = '[mat-subheader], [matSubheader]';
+
+  static with(options: SubheaderHarnessFilters = {}): HarnessPredicate<MatSubheaderHarness> {
+    return new HarnessPredicate(MatSubheaderHarness, options)
+        .addOption('text', options.text,
+            (harness, text) => HarnessPredicate.stringMatches(harness.getText(), text));
+  }
+
+  /** Gets the full text content of the list item (including text from any font icons). */
+  async getText(): Promise<string> {
+    return (await this.host()).text();
+  }
+}
+
+/**
+ * Shared behavior among the harnesses for the various `MatListItem` flavors.
+ * @docs-private
+ */
+export class MatListItemHarnessBase extends ComponentHarness {
+  private _lines = this.locatorForAll('[mat-line], [matLine]');
+  private _avatar = this.locatorForOptional('[mat-list-avatar], [matListAvatar]');
+  private _icon = this.locatorForOptional('[mat-list-icon], [matListIcon]');
+
+  /** Gets the full text content of the list item (including text from any font icons). */
+  async getText(): Promise<string> {
+    return (await this.host()).text();
+  }
+
+  /** Gets the lines of text (`mat-line` elements) in this nav list item. */
+  async getLines(): Promise<string[]> {
```

Comment:
At least to me `getLines` sounds like it would get the line elements. Maybe we should call it `getLinesText`?

Reply:
That's fair, I think we use a similar naming convention in other harnesses